### PR TITLE
Reified extensions for registering serializers in SerialModule

### LINKED ISSUE #669: Provide fluent typeOf-based API for SerialModule builders

### PR DESCRIPTION
**Pull request by qwwdfsad**
Fixes #669

### CODE REVIEW

File: runtime/commonMain/src/kotlinx/serialization/modules/PolymorphicModuleBuilder.kt [hunk 1]
```diff
@@ -51,7 +51,7 @@ public class PolymorphicModuleBuilder<Base : Any> internal constructor(
      * @see addSubclass
      */
     @ImplicitReflectionSerializer
-    public inline fun <reified T : Base> addSubclass(): Unit = addSubclass(T::class, T::class.serializer())
+    public inline fun <reified T : Base> addSubclass(): Unit = addSubclass(T::class, serializer())
```
Comment (sandwwraith) [inline runtime/commonMain/src/kotlinx/serialization/modules/PolymorphicModuleBuilder.kt]: maybe better rename this function to `subclass` alone:

```kotlin
polymorphic(Any::class, PolyBase::class) {
  subclass<PolyDerived>() // Looks more DSL-ish
}
```

File: runtime/commonMain/src/kotlinx/serialization/modules/SerialModuleBuilders.kt [hunk 2]
```diff
@@ -23,6 +23,13 @@ public fun <T : Any> serializersModuleOf(kClass: KClass<T>, serializer: KSeriali
 public inline fun <reified T : Any> serializersModule(serializer: KSerializer<T>): SerialModule =
     serializersModuleOf(T::class, serializer)
 
+/**
+ * Shortcut for [serializersModuleOf] function with type parameter.
+ */
+@ImplicitReflectionSerializer
+public inline fun <reified T : Any> serializersModule(): SerialModule =
```
Comment (sandwwraith) [inline runtime/commonMain/src/kotlinx/serialization/modules/SerialModuleBuilders.kt]: I'm not sure this is useful. Why would one want to use a module of a single default serializer? They wouldn't choose `@ContextualSerialization` in the first place.

File: runtime/commonMain/src/kotlinx/serialization/modules/PolymorphicModuleBuilder.kt (hunk 1 above)
Comment (qwwdfsad) [reply runtime/commonMain/src/kotlinx/serialization/modules/PolymorphicModuleBuilder.kt]: Great catch!

File: runtime/commonMain/src/kotlinx/serialization/modules/SerialModuleBuilders.kt (hunk 2 above)
Comment (qwwdfsad) [reply runtime/commonMain/src/kotlinx/serialization/modules/SerialModuleBuilders.kt]: For the same reason we have `serializersModule(serializer: KSerializer<T>)` and `serializersModuleOf(kClass: KClass<T>, serializer: KSerializer<T>)`

Comment (sandwwraith) [reply runtime/commonMain/src/kotlinx/serialization/modules/SerialModuleBuilders.kt]: Functions you've mentioned have explicit `serializer` parameter, i.e. user-defined one

Comment (qwwdfsad) [reply runtime/commonMain/src/kotlinx/serialization/modules/SerialModuleBuilders.kt]: Good point, I'll remove this one.
Do you have any other comments regarding this PR or I can merge it?

File: runtime/commonMain/src/kotlinx/serialization/modules/PolymorphicModuleBuilder.kt [hunk 3]
```diff
@@ -51,7 +51,7 @@ public class PolymorphicModuleBuilder<Base : Any> internal constructor(
      * @see addSubclass
      */
     @ImplicitReflectionSerializer
-    public inline fun <reified T : Base> addSubclass(): Unit = addSubclass(T::class, T::class.serializer())
+    public inline fun <reified T : Base> subclass(): Unit = addSubclass(T::class, serializer())
```
Comment (sandwwraith) [inline runtime/commonMain/src/kotlinx/serialization/modules/PolymorphicModuleBuilder.kt]: It's likely someone already used this function, so we'd better provide deprecation alias.

In other parts, PR is ok (modulo function removing we've already discussed)

**Pull request merged**